sql/table_cache.cc

```diff
@@ -129,6 +129,26 @@ void Table_cache::free_all_unused_tables() {
   }
 }
 
+void Table_cache::free_old_unused_tables(time_point cutpoint) {
+  assert_owner();
+
+  if (!m_unused_tables) return;
+  std::vector<TABLE *> old_tables;
+
+  for (TABLE *t = m_unused_tables;;) {
+    if (should_be_evicted(t->last_accessed, cutpoint)) {
+      old_tables.push_back(t);
```

Comment:
Likewise. Why collecting tables first in a custom container and delete later?

Reply:
We have a circular double-linked list, whose head can be removed and reaching the head is our stop condition.